src/eckit/geo/Iterator.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+
+#pragma once
+
+#include <type_traits>
+
+#include "eckit/geo/Point.h"
+
+
+namespace eckit::geo {
+class Grid;
```

Comment:
Unused forward declaration eckit::geo::Grid

Reply:
This is a convenience declaration to make inherited classed access in their declarations; Indeed I don't have to do it this way, but it is saving 3 (possibly 4 in the future) fwd declarations down the hierarchy